src/webrtc/callFeed.ts

```diff
@@ -76,20 +64,28 @@ type EventHandlerMap = {
     [CallFeedEvent.Disposed]: () => void;
 };
 
-export class CallFeed extends TypedEventEmitter<CallFeedEvent, EventHandlerMap> {
-    public feedId: string;
-    public readonly userId: string;
-    public readonly deviceId: string | undefined;
-    public purpose: SDPStreamMetadataPurpose;
+export abstract class CallFeed extends TypedEventEmitter<CallFeedEvent, EventHandlerMap> {
+    public abstract readonly id?: string;
+    public abstract readonly streamId?: string;
```

Comment:
Since we also have the stream object in the `CallFeed`, we should not set the stream ID as an extra property. We could make them accessible via a `getter`. This has the advantage if the stream change to not having to update this property too.